tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -833,65 +898,26 @@ def _test_many_partitions(self, compacted):
+                                          topic_names, n_partitions,
+                                          scale.expect_bandwidth / 60)
 
-            def progress_check():
-                # Explicit wait for consumer group, because we might have e.g.
-                # just restarted the cluster, and don't want to include that
-                # delay in our throughput-driven timeout expectations
-                self.logger.info(f"Checking repeater group is ready...")
-                repeater.await_group_ready()
-
-                t = repeater_await_bytes / scale.expect_bandwidth
-                self.logger.info(
-                    f"Waiting for {repeater_await_msgs} messages in {t} seconds"
-                )
-                t1 = time.time()
-                repeater.await_progress(repeater_await_msgs, t)
-                t2 = time.time()
-
-                # This is approximate, because await_progress isn't returning the very
-                # instant the workers hit their collective target.
-                self.logger.info(
-                    f"Wait complete, approx bandwidth {(repeater_await_bytes / (t2-t1))/(1024*1024.0)}MB/s"
-                )
-
-            progress_check()
-
-            self.logger.info(f"Entering single node restart phase")
-            self._single_node_restart(scale, topic_names, n_partitions)
-            progress_check()
-
-            self.logger.info(f"Entering restart stress test phase")
-            self._restart_stress(scale, topic_names, n_partitions,
-                                 progress_check)
-
-            self.logger.info(
-                f"Post-restarts: checking repeater group is ready...")
-            repeater.await_group_ready()
-
-            # Done with restarts, now do a longer traffic soak
-            self.logger.info(f"Entering traffic soak phase")
-            soak_await_bytes = 100E9
-            if not self.redpanda.dedicated_nodes:
-                soak_await_bytes = 10E9
-
-            soak_await_msgs = soak_await_bytes / repeater_msg_size
-            t1 = time.time()
-            initial_p, _ = repeater.total_messages()
-            try:
-                repeater.await_progress(
-                    soak_await_msgs, soak_await_bytes / scale.expect_bandwidth)
-            except TimeoutError:
-                t2 = time.time()
-                final_p, _ = repeater.total_messages()
-                bytes_sent = (final_p - initial_p) * repeater_msg_size
-                expect_mbps = scale.expect_bandwidth / (1024 * 1024.0)
-                actual_mbps = (bytes_sent / (t2 - t1)) / (1024 * 1024.0)
-                self.logger.error(
-                    f"Expected throughput {expect_mbps:.2f}, got throughput {actual_mbps:.2f}MB/s"
-                )
-                raise
+        # non-txn background traffic
+        repeater_msg_size = 16384
+        with repeater_traffic(context=self._ctx,
```

Comment:
It's a bit odd to pick out the restart stress portion of the test and only run that bit with transactions.

The approach for compaction is to have a top level set of `repeater_kwargs`, and two different test cases (compacted vs. non) -- can we do the same for transactions?  Maybe rather than each feature independently, we should have just two variants of the test: vanilla (no compaction or transactions) and "fully loaded" (compaction, transactions enabled).

I think that'll be useful in identifying which failures result from issues in the primary data path, vs. which ones relate to particular optional features.